### PR TITLE
fix method ambiguity for bandwidth, (fixes #345, fixes #346)

### LINKED ISSUE #345: Method ambiguity warning
With both ApproxFun 0.2.1 and master, I get:
```
INFO: Recompiling stale cache file /Users/stevenj/.julia/lib/v0.4/ApproxFun.ji for module ApproxFun.
WARNING: New definition 
    bandwidth(ApproxFun.SubBandedMatrix, Any) at /Users/stevenj/.julia/v0.4/ApproxFun/src/Operators/SubMatrix.jl:21
is ambiguous with: 
    bandwidth(AbstractArray{T<:Any, 2}, Integer) at /Users/stevenj/.julia/v0.4/BandedMatrices/src/BandedMatrices.jl:71.
To fix, define 
    bandwidth(ApproxFun.SubBandedMatrix, Integer)
```
cc @rpestourie

### LINKED ISSUE #346: 2d derivative operator is broken:
With either 0.2.1 or master, the following code
```jl
using ApproxFun
d1 = Space([0,1])
d2 = Space([0,2])
d = d1 * d2
D = Derivative(d, [0,1]);
f = Fun((x,y) -> sin(x) * cos(y), d)
D * f
```
gives
```
LoadError: BoundsError: attempt to access 325-element Array{Float64,1}:
 -0.333206
  0.0     
  0.0     
  0.0     
  0.0     
  0.0     
  0.0     
  ...
```
cc @rpestourie

### PR DESCRIPTION
**Pull request by stevengj**
This fixes #345, and in doing so fixes #346 as well.

### CODE REVIEW

File: test/OperatorTest.jl [hunk 1]
```diff
@@ -180,3 +180,11 @@ A=[dirichlet(S);Derivative(S)^2 - exp(im*x)]
 QR=qrfact(A)
 
 @test norm((A\[1.])-(QR\[1.]))<100eps()
+
+# 2d derivative (issue #346)
+let d = Space([0,1]) * Space([0,2]),
+    D = Derivative(d, [0,1]),
+    f = Fun((x,y) -> sin(x) * cos(y), d),
+    fd = Fun((x,y) -> -sin(x) * sin(y), d)
+    @test (D*f)(0.2,0.3) ≈ fd(0.2,0.3)
```
Comment (stevengj) [inline test/OperatorTest.jl]: Note that what I really wanted to do was `@test norm(D*f - fd) < 100*eps()`, but the expression `D*f - fd` gives
```
ERROR: AssertionError: iskronsumop(A)
 in schurfact at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/factorizations.jl:197
 in schurfact at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/factorizations.jl:215
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:177
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:135
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:179
 in \ at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:187
 in defaultcoefficients at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Space.jl:351
 in coefficients at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Fun.jl:46
 in call at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/constructors.jl:66
 in - at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Fun.jl:197
```
Aren't we supposed to be able to subtract functions and derivatives?

**Comment by dlfivefifty** [comment]
Thanks!  Multivariate/PDE support is really a WIP and about to undergo a major revamp.  So I wouldn't try to use it at the moment far beyond the examples.

**Comment by dlfivefifty** [comment]
Did you intend for this merge to go in to development and not master?  In your issue, you mentioned the bug being in master.  (I'm under the impression the bug was fixed in development already.)

**Comment by stevengj** [comment]
Ah, I forgot that you use a separate development branch.

**Comment by stevengj** [comment]
Yes, looks like it's already fixed in development.

**Comment by dlfivefifty** [comment]
Yes, the intention is that master is always useable, where development may at times be experimental.

**Comment by dlfivefifty** [comment]
I can tag a new version over the weekend.

**Comment by stevengj** [comment]
Thanks!

**Comment by dlfivefifty** [comment]
Just tagged 0.2.2

**Pull request merged**